soc/st/stm32/stm32g0x/Kconfig.soc

```diff
@@ -73,3 +73,8 @@ config SOC
 	default "stm32g0b0xx" if SOC_STM32G0B0XX
 	default "stm32g0b1xx" if SOC_STM32G0B1XX
 	default "stm32g0c1xx" if SOC_STM32G0C1XX
+
+config SOC_FLASH_PREFETCH_ENABLE
+	bool "FLASH prefetch on STM32G0B0/G0B1/G0C1 series"
+	default n if SOC_STM32G0B0XX || SOC_STM32G0B1XX || SOC_STM32G0C1XX
+	default y
```

Comment:
There is already a `STM32_FLASH_PREFETCH` config option that does the very same thing. I think it could be reused. Moreover, it has a `STM32_` prefix that better suit a config option that only relate to STM32 SoCs.